### PR TITLE
Samples maintenance

### PR DESCRIPTION
**Pull request by domesticmouse**
Hey all, I just made a pass through the samples looking at warnings and following linter suggestions. There are some deprecation warnings in the `web` samples that I'm unsure how to fix. What is our policy on maintaining these samples? I'd advocate that they are either kept at latest revisions and warning free, or they get deleted/archived somewhere else.

Thoughts?

### CODE REVIEW

**Comment by farr64** [comment]
Please:  Keep at latest revisions and warning free.

Thanks for all the work involved in doing so.

**Comment by domesticmouse** [comment]
PTAL @RedBrogdon @johnpryan

**Comment by johnpryan** [comment]
I vote to leave `web` directory as-is for now, it's likely we will have to either upgrade them to NNBD or drop them eventually.

File: place_tracker/lib/place_map.dart [hunk 1]
```diff
@@ -47,7 +50,7 @@ class PlaceMapState extends State<PlaceMap> {
 
   LatLng _lastMapPosition;
 
-  Map<Marker, Place> _markedPlaces = Map<Marker, Place>();
+  final Map<Marker, Place> _markedPlaces = <Marker, Place>{};
```
Comment (johnpryan) [inline place_tracker/lib/place_map.dart:53]: nit: The type can be [inferred](https://dart.dev/guides/language/sound-dart#type-inference) with just `final _markedPlaces = <Marker, Place>{};` or `final Map<Marker, Place> _markedPlaces = {};`

File: veggieseasons/lib/data/preferences.dart [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Preferences extends Model {
 
   int _desiredCalories = 2000;
 
-  Set<VeggieCategory> _preferredCategories = Set<VeggieCategory>();
+  final Set<VeggieCategory> _preferredCategories = <VeggieCategory>{};
```
Comment (johnpryan) [inline veggieseasons/lib/data/preferences.dart:23]: Same here

File: place_tracker/lib/place_map.dart (hunk 1 above)
Comment (domesticmouse) [reply place_tracker/lib/place_map.dart:53]: I believe the current style (at least as enforced by the lint checks) is to turn method locals into `var`s, but to leave class fields as fully typed.

File: veggieseasons/lib/data/preferences.dart (hunk 2 above)
Comment (domesticmouse) [reply veggieseasons/lib/data/preferences.dart:23]: Yup, this is also a class field

File: place_tracker/lib/place_map.dart (hunk 1 above)
Comment (johnpryan) [reply place_tracker/lib/place_map.dart:53]: I would vote for the second option then.

File: veggieseasons/lib/data/preferences.dart (hunk 2 above)
Comment (johnpryan) [reply veggieseasons/lib/data/preferences.dart:23]: same

File: place_tracker/lib/place_map.dart (hunk 1 above)
Comment (domesticmouse) [reply place_tracker/lib/place_map.dart:53]: Wanna spin up a new PR with your suggestion? This PR has landed and I've already deleted the source repo =)

**Pull request merged**